### DISCUSSION TITLE
Bicycle routing compared to OsmAnd

### DISCUSSION THREAD

**Discussion by DirkDuesentrieb** [organicmaps/organicmaps#2374] [Ideas]
Hi,
I'm using OsmAnd for bicycle navigation for several years now but wanted to give Organic Maps a try. It is cool how light and snappy the app is. Finding a destination is quicker compared to OsmAnd and the maps look good.
BUT: Using navigation here in Hamburg, does not work for me - the resulting cycle routes are following main streets, which are mostly loud and dangerous. OsmAnd allows to select a navigation strategy, it goes from "offroad" to "quickest" to make all cyclists happy. The default "balanced" profile creates very good ways through the city, avoiding stressful roads. Organic Maps seems to use the "quickest path" strategy and has IMHO no option to change this. 

Is someone else using Organic Maps and/or OsmAnd for bicycle navigation and can share her/his experience?

**Comment by biodranik** [comment]
Thanks for the feedback! We plan to add #2153 (the Alternate routes support) for all means of transport. For now, did you try the "avoid highways" option? You may also add an intermediate stop after building a route to avoid some busy streets.

**Comment by DirkDuesentrieb** [comment]
Thanks for the quick answer. The alternate routes support will be great! Maybe you can clone OsmAnds routing strategies; the resulting routes are working really good for me.

**Comment by renatamg** [comment]
Hello @DirkDuesentrieb, my experience is exactly the opposite. I'm in Graz (Austria), and just installed Organic Maps and went out on a ride to test it a few days ago... While OsmAnd would have me get out of the bikeway much too soon, Organic Maps route is exactly how I always do it, delaying as much as possible the turn into a street (which is actually still officially a bikeway too). The OsmAnd way is even a bit longer than the Organic Maps one.
This is a problem I often had with OsmAnd, no matter how I configured it, it wants me to ride with cars. So I'm quite happy with Organic Maps route calculation. And indeed, it is MUCH faster and feels lighter too.

The one thing I'm missing is to explore the route before, navigate it with the height profile to know where I'll have to put more effort and so on. And separating ascend and descend would be very nice too :)

**Comment by fami-64** [comment]
Maybe by using opencyclemap data the routes will be more suitable for cyclists

**Reply by biodranik** [reply]
What do you mean? OpencycleMap also takes all routing data from OpenStreetMap. Right?

**Reply by fami-64** [reply]
Yes, but opencyclemap includes additional information such as bike routes and bike paths. This would be useful to calculate a route adapted to cyclists.
For example, for the route calculation between Urcuit and Bayonne (France), it doesn't propose the V81 bike path but the main road which is dangerous. It does not propose either to go along the railroad by a small road which is much more adapted to bikes and not much longer.
Similarly, between Urcuit and Hasparren (France), the route goes along the main road which is very busy with cars, and not along the small parallel roads.

Translated with www.DeepL.com/Translator (free version)

**Reply by fami-64** [reply]
Sample from Urcuit to Hasparren through the small roads : https://www.francevelotourisme.com/itineraire/gps?type=recommande&points=%5B%7B%22coordinates%22%3A%5B-1.3372408%2C43.4816866%5D%2C%22label%22%3A%22Chemin%20Asserol%20%28Urcuit%29%22%2C%22input%22%3A0%7D%2C%7B%22coordinates%22%3A%5B-1.3049537999999998%2C43.38388%5D%2C%22label%22%3A%22Hasparren%20%2864240%29%2C%20Pyr%5Cu00e9n%5Cu00e9es-Atlantiques%2C%20Nouvelle-Aquitaine%2C%20France%22%2C%22input%22%3A1%7D%5D

**Reply by biodranik** [reply]
We can improve our algorithms. When alternate routes will be supported, Organic Maps routes will be better. Now you can workaround it by adding intermediate stops. 

CC @vng

**Reply by fami-64** [reply]
Thank you for your answer, but it doesn't seem very appropriate when you don't know the route, which is often my case.
As I like your approach, is it to import a GPX file (I would then have prepared the route with another platform)?

**Reply by biodranik** [reply]
You can import any KML or KMZ file. GPX can be converted to KML. GPX import is not ready yet.
